### PR TITLE
cluster-autoscaler

### PR DESCRIPTION
**Pull request by resmaa**
**3.1 Release Feature**

Cluster aAutoscaler
PCP-630

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1052#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 1]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
 ## Running Clusters through Edit Node Pool
 <InfoBox>
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1019]: This does not need to be in an info box, it should be the heading's introductory paragraph.

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 2]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
 ## Running Clusters through Edit Node Pool
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1018]:
```suggestion
## Edit Node Pool

```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 3]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+</InfoBox>
+
+**Master Pool**
 
 |**Parameter**     | **Description**|
 |------------------|---------------|
 |**Name**          |A descriptive name for the node pool.|
 |**Size**          |Number of VMs to be provisioned for the node pool. For the master pool, this number can be 1, 3, or 5.|
-|**Allow worker capability (master pool)**|Select this option for allowing workloads to be provisioned on master nodes.|
+|**Allow worker capability (master pool)**|Select this option for allowing workloads to be provisioned on master nodes.|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1029]:
```suggestion
|**Allow worker capability **|Select this option for allowing workloads to be provisioned on master nodes.|
```
We are already in the context of Master Pool, so we don't need to specify master pool.

Why do we need to

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 4]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+|**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add to constrain a Pod to only run on a particular set of Node(s).
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1030]:
```suggestion
|**[Labels](/clusters/cluster-management/taints#overviewonlabels)**| Add a label to apply placement constraints on a pod, such as a node eligible for receiving the workload. 

```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 5]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+|**[Taints](/clusters/cluster-management/taints#overviewontaints)**|To set toleration to pods and allow (but do not require) the pods to schedule onto nodes with matching taints.|
 |**Instance type** |Select the AWS instance type to be used for all nodes in the node pool.|
+|**Availability Zones**| Choose one or more availability zones. Palette provides fault tolerance to guard against failures like hardware failures, network failures, etc. by provisioning nodes across availability zones if multiple zones are selected.|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1033]:
```suggestion
|**Availability Zones**| Choose one or more availability zones. Palette provides fault tolerance to guard against hardware failures, network failures, etc., by provisioning nodes across availability zones if multiple zones are selected.|
```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 6]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+|**Disk Size**|Give the required storage size|
+
+**Worker Pool**
+
+|**Parameter**     | **Description**|
+|------------------|---------------|
+|**Name**          |A descriptive name for the node pool.|
+|**Enable Autoscaler**|You can enable the autoscaler, by toggling the **Enable Autoscaler** button. Autoscaler scale up and scale down the resources between the set minimum and maximum number of nodes to optimize resource utilization.|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1041]:
```suggestion
|**Enable Autoscaler**|You can enable the autoscaler, by toggling the **Enable Autoscaler** button. Autoscaler scales up and down resources between the defined minimum and the maximum number of nodes to optimize resource utilization.|
```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 7]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+||Set the scaling limit by setting the **Minimum Size** and **Maximum Size**, as per the workload the number of nods will scale up from minimum set value to maximum set value and the scale down from maximum set value to minimum set value|
+|**Size**          |Number of VMs to be provisioned for the node pool.|
 |**Rolling Update**| There are two choices of Rolling Update:
 |                  |**Expand First**: Launches the new node and then shut down the old node.
-|                  |**Contract First**: Shut down the old node first and then launches the new node.
-|**Availability Zones**| Choose one or more availability zones. Palette provides fault tolerance to guard against failures like hardware failures, network failures, etc. by provisioning nodes across availability zones if multiple zones are selected.|
-
+|                  |**Contract First**: Shut down the old node first and then launches the new node.
+|**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add to constrain a Pod to only run on a particular set of Node(s).
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1047]:
```suggestion
|**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add a label to apply placement constraints on a pod, such as a node eligible for receiving the workload.

```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 8]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+|**[Taints](/clusters/cluster-management/taints#overviewontaints)**|To set toleration to pods and allow (but do not require) the pods to schedule onto nodes with matching taints.|
+|**Instance type** |Select the AWS instance type to be used for all nodes in the node pool.|
+|**Availability Zones**| Choose one or more availability zones. Palette provides fault tolerance to guard against failures like hardware failures, network failures, etc. by provisioning nodes across availability zones if multiple zones are selected.|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1050]:
```suggestion
|**Availability Zones**| Choose one or more availability zones. Palette provides fault tolerance to guard against hardware failures, network failures, etc., by provisioning nodes across availability zones if multiple zones are selected.|
```

File: content/docs/04-clusters/01-public-cloud/01-aws.md [hunk 9]
```diff
@@ -1018,19 +1018,37 @@ sigs.k8s.io/cluster-api-provider-aws/cluster/[ClusterName] = owned
...
+|**Disk Size**|Give the required storage size|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/01-aws.md:1051]:
```suggestion
|**Disk Size**|Provide the required storage size|

```

File: content/docs/04-clusters/01-public-cloud/03-azure/02-create-azure-cluster.md [hunk 10]
```diff
@@ -99,18 +99,36 @@ If the Palette [cloud account](/clusters/public-cloud/azure#creatinganazurecloud
...
 7. Configure the master and worker node pools. A master and a worker node pool are configured by default.
+  
+  **Master Pool Configuration** 
 
     |**Parameter**| **Description**|
     |-------------|----------------|    
     |**Name** | A descriptive name for the node pool|
     |**Size** | Number of nodes to be provisioned for the node pool. For the master pool, this number can be 1, 3, or 5|
     |**Allow worker capability (master pool)**| To allow workloads to be provisioned on master nodes|
+    |**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add to constrain a Pod to only run on a particular set of Node(s).
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/03-azure/02-create-azure-cluster.md:110]:
```suggestion
    |**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add a label to apply placement constraints on a pod, such as a node eligible for receiving the workload.

```

File: content/docs/04-clusters/01-public-cloud/03-azure/02-create-azure-cluster.md [hunk 11]
```diff
@@ -99,18 +99,36 @@ If the Palette [cloud account](/clusters/public-cloud/azure#creatinganazurecloud
...
+    |-------------|----------------| 
+    |**Node pool Name** | A descriptive name for the worker node pool| 
+    |**Enable Autoscaler**|You can enable the autoscaler, by toggling the **Enable Autoscaler** button. Autoscaler scale up and scale down the resources between the set minimum and maximum number of nodes to optimize resource utilization.|
+    ||Set the scaling limit by setting the **Minimum Size** and **Maximum Size**, as per the workload the number of nods will scale up from minimum set value to maximum set value and the scale down from maximum set value to minimum set value|
+    |**Pool Size**|Number of nodes in the pool.|
     |**Rolling Updates**| There are two choices of Rolling Update.|
     ||**Expand First**: Launches the new node and then shut down the old node.|
-    ||**Contract First**: Shut down the old node first and then launches the new node.|
+    ||**Contract First**: Shut down the old node first and then launches the new node.|
+    |**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add to constrain a Pod to only run on a particular set of Node(s).
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/03-azure/02-create-azure-cluster.md:127]:
```suggestion
    |**[Labels](/clusters/cluster-management/taints#overviewonlabels)**|Add a label to apply placement constraints on a pod, such as a node eligible for receiving the workload.

```

File: content/docs/04-clusters/01-public-cloud/05-gcp.md [hunk 12]
```diff
@@ -197,22 +197,38 @@ Performed the steps below to provision a new GCP cluster:
...
+  |**Instance Type**|Select the Azure instance type to be used for all the nodes in the pool|
+  |**Availability Zones**|Choose one or more availability zones. Palette provides fault <br /> tolerance to guard against failures like hardware failures, network failures, etc. by provisioning nodes <br /> across availability zones if multiple <br /> zones are selected.|
+  |**Disk Size**|Storage disk size in GB to be attached to the node.|
+
+**Worker Pool Configuration** 
+
+  |**Parameter**| **Description**|
+  |-------------|----------------| 
+  |**Node pool Name** | A descriptive name for the worker node pool| 
+  |**Enable Autoscaler**|You can enable the autoscaler, by toggling the **Enable Autoscaler** button. Autoscaler scale up and scale down the resources between the set minimum and maximum number of nodes to optimize resource utilization.|
```
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/01-public-cloud/05-gcp.md:220]:
```suggestion
  |**Enable Autoscaler**|You can enable the autoscaler, by toggling the **Enable Autoscaler** button. Autoscaler scales up and down resources between the defined minimum and the maximum number of nodes to optimize resource utilization.|
```

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Feedback and suggestions below.

**Review by karl-cardenas-coding — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1052#pullrequestreview-1225122931" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1052#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by karl-cardenas-coding** [comment]
@resmaa, I'm gonna close this PR. I went ahead and carried over the changes in a new PR #1061 that is based of the release branch. That gets us around the merge conflicts.

**Pull request closed without merging**